opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/sort/SortQueryBuilderTest.java

```diff
@@ -25,6 +28,62 @@ void build_sortbuilder_from_reference() {
     assertNotNull(sortQueryBuilder.build(DSL.ref("intV", INTEGER), Sort.SortOption.DEFAULT_ASC));
   }
 
+  @Test
+  void build_sortbuilder_from_nested_function() {
+    assertNotNull(
+        sortQueryBuilder.build(
+            DSL.nested(DSL.ref("message.info", STRING)),
+            Sort.SortOption.DEFAULT_ASC
+        )
+    );
+  }
+
+  @Test
+  void build_sortbuilder_from_nested_function_with_path_param() {
+    assertNotNull(
+        sortQueryBuilder.build(
+            DSL.nested(DSL.ref("message.info", STRING), DSL.ref("message", STRING)),
+            Sort.SortOption.DEFAULT_ASC
+        )
+    );
+  }
+
+  @Test
+  void nested_with_too_many_args_throws_exception() {
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> sortQueryBuilder.build(
+            DSL.nested(
+                DSL.ref("message.info", STRING),
+                DSL.ref("message", STRING),
+                DSL.ref("message", STRING)
+            ),
+            Sort.SortOption.DEFAULT_ASC
+        )
+    );
+  }
+
+  @Test
+  void nested_with_invalid_arg_type_throws_exception() {
```

Comment:
do we want to add a test case to validate that `nested()` returns an IllegalArgumentException?

Reply:
Already have one [here](https://github.com/opensearch-project/sql/blob/84fab78dce84f89befa6012acc55792c2078892f/opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/sort/SortQueryBuilderTest.java#L52-L64)